### ISSUE TITLE
Question about how to generate backprojection image using ground truth data

### ISSUE THREAD

**Issue by TSAI-CHANCHANG** [nianticlabs/monodepth2#246]
Thank you for your work!

### **Background**
In the stage of loss computing, the system needs to generate a back-projection image.
Function: `generate_images_pred`
Input: 
1. the depth prediction of target image (or we can call it image `0`);
2. pre-computed intrinsics K;
3. the RGB source image (or we can call it RGB image `0+1`)
4. the relative pose prediction from target to source (or we can call it `0->1`)
Output:
A back-projection image `1->0`

### **Motivation**
Now I have 2 RGB images with their **ground-truth pose** and **ground-truth depth map** (these 2 images are the k and k+1 frame in one video sequence).
And I want to use the following things to generate a back-projection image:
1. gt-depth of image `0`;
2. intrinsics `K`
3. the RGB image `1`;
4. the gt-relative pose;

### **Question**
I write the code to generate back-projection images (can be checked in the last part of this issue).
The code seems to generate images well. But if I select other images as source image (for example, instead of picking up `k+1`, now I pick up `k+10`, `k+20`, `k+50` as my source image), the generated image `50->0` will show larger and larger residuals from the original view.
If I understand correctly, under the right configuration, the function `generate_images_pred` should generate an image with no residual (even the edge part of the generated is noisy, the rest of the image pixels should in the same place as the original view)
generate_gt_50->0

![image](https://user-images.githubusercontent.com/26506187/92912140-78dedd80-f464-11ea-94b8-77773b0e3258.png)
![image](https://user-images.githubusercontent.com/26506187/92912507-d4a96680-f464-11ea-90bd-29ea8d89c1b7.png)
![generate_gt_50->0](https://user-images.githubusercontent.com/26506187/92914939-0de2d600-f467-11ea-8117-778c144d4756.png)

So I am curious about what causes this.

**And several sub-questions I want to ask** 
1. How do you normalize your ground truth data of KITTI when evaluating the depth loss? I think maybe the wrong way to preprocess my ground truth depth data cause this problem
2. What are the units of ground truth data of KITTI (which can be downloaded from [http://www.cvlibs.net/datasets/kitti/eval_depth.php?benchmark=depth_prediction](url))?

### **Things for check**
**depth image in my dataset**
Depth: frame-XXXXXX.depth.png (depth in millimeters, 16-bit, PNG, invalid depth is set to 65535).
**K**
```python
self.K = np.array([[585/640, 0, 320/640, 0],
                           [0, 585/480, 240/480, 0],
                           [0, 0, 1, 0],
                           [0, 0, 0, 1]], dtype=np.float32)
```
**code**
This is the code I program for generating images using gt-data.
https://github.com/TSAI-CHANCHANG/monodepth2-1/blob/master/generate_image_use_gt.py
the key part is as follows:

```python
with torch.no_grad():
    for inputs in dataloader:
        for key, ipt in inputs.items():
            inputs[key] = ipt.cuda()

        # step 1: input ground truth depth
        outputs[("disp", 0)] = inputs["depth_gt"] / 65535
                
        # step 2: input the ground truth pose of 2 image
        # then calculate the relative pose
        fStr1 = "frame-{:06d}.pose.txt".format(inputs["index"].item())
        gtPosePath1 = os.path.join("/content/drive/My Drive/monodepth2/splits/7scenes/chess/seq-01", "poses", fStr1)
        gtPose1 = np.loadtxt(gtPosePath1).reshape(4, 4)
        fStr2 = "frame-{:06d}.pose.txt".format(inputs["index"].item()+opt.frame_ids[1])
        print("fStr1 = {}".format(fStr1))
        print("fStr2 = {}".format(fStr2))
        gtPosePath2 = os.path.join("/content/drive/My Drive/monodepth2/splits/7scenes/chess/seq-01", "poses", fStr2)
        gtPose2 = np.loadtxt(gtPosePath2).reshape(4, 4)
        gtRelativePose = calRelativePose(gtPose1, gtPose2)

        outputs[("cam_T_cam", 0, opt.frame_ids[1])] = torch.from_numpy(gtRelativePose.reshape(1, 4, 4).astype(np.float32)).cuda()

        # step 3: using function "generate_images_pred" to generate backprojection image
        self.generate_images_pred(inputs, outputs)
        index = inputs["index"].cpu().item()
                
        img_2 = transforms.ToPILImage()(outputs[("color", opt.frame_ids[1], 0)].squeeze().cpu()).convert('RGB')
        img_2.save("/content/drive/My Drive/monodepth2/generate_gt.jpg") 
```
